revolutionpi/analog_pin.go

```diff
@@ -127,3 +138,57 @@ func (pin *analogPin) isAnalogOutput() bool {
 func (pin *analogPin) isAnalogInput() bool {
 	return pin.Address >= pin.inputOffset && pin.Address < pin.inputOffset+8
 }
+
+func getAnalogOutputRange(val byte, name string) (analogInfo, error) {
+	switch val {
+	case 0:
+		return analogInfo{}, fmt.Errorf("pin %s is not configured for analog write", name)
+	case 1: // 0 to 5000 mV
+		return analogInfo{min: 0, max: 5000, isCurrent: false}, nil
+	case 2: // 0 to 10000 mV
+		return analogInfo{min: 0, max: 10000, isCurrent: false}, nil
+	case 3: // -5000 to 5000 mV
+		return analogInfo{min: -5000, max: 5000, isCurrent: false}, nil
+	case 4: // -10000 to 10000 mV
+		return analogInfo{min: -10000, max: 10000, isCurrent: false}, nil
+	case 5: // 0 to 5500 mV
+		return analogInfo{min: 0, max: 5500, isCurrent: false}, nil
+	case 6: // 0 to 11000 mV
+		return analogInfo{min: 0, max: 11000, isCurrent: false}, nil
+	case 7: // -5500 to 5500 mV
+		return analogInfo{min: -5500, max: 5500, isCurrent: false}, nil
+	case 8: // -11000 to 11000 mV
+		return analogInfo{min: -11000, max: 11000, isCurrent: false}, nil
+	case 9: // 4 to 20 mA
+		return analogInfo{min: 4000, max: 20000, isCurrent: true}, nil
+	case 10: // 0 to 20 mA
+		return analogInfo{min: 0, max: 20000, isCurrent: true}, nil
+	case 11: // 0 to 24 mA
+		return analogInfo{min: 0, max: 24000, isCurrent: true}, nil
+	default:
+		return analogInfo{}, fmt.Errorf("invalid output range received, got %v", val)
+	}
+}
+
+func getAnalogInputRange(val byte) (analogInfo, error) {
+	switch val {
+	case 1: // -10000 to 10000 mV
+		return analogInfo{min: -10000, max: 10000, isCurrent: false}, nil
+	case 2: // 0 to 10000 mV
+		return analogInfo{min: 0, max: 10000, isCurrent: false}, nil
+	case 3: // 0 to 5000 mV
+		return analogInfo{min: 0, max: 5000, isCurrent: false}, nil
+	case 4: // -5000 to 5000 mV
+		return analogInfo{min: -5000, max: 5000, isCurrent: false}, nil
+	case 5: // 0 to 20 mA
+		return analogInfo{min: 0, max: 20000, isCurrent: true}, nil
+	case 6: // 0 to 24 mA
+		return analogInfo{min: 0, max: 24000, isCurrent: true}, nil
+	case 7: // 4 to 20 mA
+		return analogInfo{min: 4000, max: 20000, isCurrent: true}, nil
+	case 8: // -25 to 25 mA
+		return analogInfo{min: -25000, max: 25000, isCurrent: true}, nil
```

Comment:
i would prefer the min/max be mA instead of micro to match the voltage

Reply:
I sorta agree, but I want to make sure that the min/max units are the same units as the Value, and I do not want to lose any current values that are below the mA scale